### PR TITLE
Add NULL checks to GetClassWithPossibleAV

### PR DESCRIPTION
**Pull request by mikem8361**
Fixes the SIGSEGV in the Alpine SOS tests.

### CODE REVIEW

**Review by jkotas — APPROVED** [review]
Thanks

File: src/coreclr/vm/methodtable.inl [hunk 1]
```diff
@@ -29,7 +29,12 @@ FORCEINLINE PTR_EEClass MethodTable::GetClassWithPossibleAV()
     LIMITED_METHOD_DAC_CONTRACT;
 
     TADDR addr = m_pCanonMT;
-
+#ifdef DACCESS_COMPILE
+    if (addr == 0)
```
Comment (jkotas) [inline src/coreclr/vm/methodtable.inl:33]:
```suggestion
    if (addr == (TADDR)NULL)
```

File: src/coreclr/vm/methodtable.inl [hunk 2]
```diff
@@ -39,6 +44,12 @@ FORCEINLINE PTR_EEClass MethodTable::GetClassWithPossibleAV()
     {
         // pointer to canonical MethodTable.
         TADDR canonicalMethodTable = union_getPointer(addr);
+#ifdef DACCESS_COMPILE
+        if (canonicalMethodTable == 0)
```
Comment (jkotas) [inline src/coreclr/vm/methodtable.inl:48]:
```suggestion
        if (canonicalMethodTable == (TADDR)NULL)
```

**Review by lambdageek — APPROVED** [review]

**Comment by jkotas** [comment]
/ba-g eventsourceerror is a known issue, the test is disabled against https://github.com/dotnet/runtime/issues/80666

**Pull request merged**